lrlex/src/lib/parser.rs

```diff
@@ -244,9 +225,11 @@ mod test {
             Ok(_) => panic!("Broken rule parsed"),
             Err(LexBuildError {
                 kind: LexErrorKind::MissingSpace,
-                line: 2,
-                col: 1,
-            }) => (),
+                span,
```

Comment:
Might this be nicer as `span if span == Span::new(3, 3)`? Maybe not!

Reply:
Hmm, I could go either way on this, I had done pattern guards a few places, but I ran into issues where the test was failing, and the `assert_eq!` produced output about the failure, where a pattern guard just wouldn't match, so I'd moved them all to assertions.

So i'm kind of hesitant, let me know if you still want me to try it out that way though.

Reply:
So, I have come around to your suggestion with the addition that we format the error like it used to be,
and working on an implementation of that..

Reply:
Alright, pushed that and feel that it is an improvement, but feel free to unresolve or change your mind if you feel otherwise